lxd/instance/drivers/driver_qemu.go

```diff
@@ -3097,6 +3097,19 @@ func (d *qemu) deviceBootPriorities(base int) (map[string]int, error) {
 	return sortedDevs, nil
 }
 
+// shortenedSocketPath creates an alternative path to a socketTrickery by using the file descriptor to the directoryof the socket file.
+// Used to handle paths > 107 chars.
+func (d *qemu) shortenedSocketPath(originalSockPath string, openFiles *[]*os.File, fdFiles *[]*os.File) (string, error) {
```

Comment:
Isn't the slice always passed as reference? I am wondering if you need the extra pointer here for the two args.

Reply:
I've used a pointer to a slice here to be sure that when we are appending to it the resulting slice is stored back at the original location. I understand a Go slice is an internal struct that is copied when passed by value, but which internally has a pointer to a backing array. This is why updating the entry of a slice that is passed by value into a function can be seen in the original slice variable by the caller.

However if `append` internally creates a new array, the pointer will only be updated in the returned slice value from append, which unless assigned back to the original slice value will not be reflected in the caller.

This would have unpredictable consequences depending on when Go decides to expand the slice using a new backing array.

See https://go.dev/play/p/4u2zb7QVy_8

Reply:
Maybe I could add a comment so it is clearer for anyone who reads this secction from now on?

Reply:
@tomponline ah yes, thanks for the insights, quite a dangerous suggestion then from my side. 

@hamistao I guess we don't require a comment here.